### PR TITLE
Copy edit uVisor documents

### PR DESCRIPTION
**Pull request by AnotherButler**
Update uVisor documents for active voice, consistency, grammar and spelling

### CODE REVIEW

File: docs/core/PORTING.md [hunk 1]
```diff
@@ -241,7 +227,7 @@ The symbols that you can specify here are listed in the table below.
```
Comment (AnotherButler) [inline docs/core/PORTING.md]: Should all of these TODOs be here?

File: docs/core/PORTING.md [hunk 2]
```diff
@@ -570,58 +556,58 @@ As shown in the snippet above, the uVisor needs the following regions:
...
+        It hosts pages the uVisor secure allocator allocates. If the uVisor shares the SRAM with the OS/app, it must be immediately after the uVisor BSS section. Otherwise, it must be the first SRAM region after the VTOR relocation section.
       </td>
     </tr>
     <tr>
       <td>
         <code>.uvisor.secure</code>
       </td>
       <td>
-        It contains constant data in flash that describes the configuration of the secure boxes. It comprises the configuration tables (<code>.keep.uvisor.cfgtbl</code>) and the pointers to them (<code>.keep.uvisor.cfgtbl_ptr[_first]</code>), which are used by uVisor for box enumeration.
+        It contains constant data in flash that describes the configuration of the secure boxes. It comprises the configuration tables (<code>.keep.uvisor.cfgtbl</code>) and the pointers to them (<code>.keep.uvisor.cfgtbl_ptr[_first]</code>), which uVisor uses for box enumeration.
```
Comment (AnotherButler) [inline docs/core/PORTING.md]: Does `uvisor.secure` make up the configuration tables and pointers to them, or do the configuration tables and pointers make up `uvisor.secure`?

File: docs/api/manual/UseCases.md [hunk 3]
```diff
@@ -308,97 +290,97 @@ After the receiver is done with the message, it must free resources backing the
...
+Likewise, using the process stack for thread stacks is also not feasible. This is because multiple threads cannot share the same stack if they are not guaranteed to stack and unstack in a consistent order (first in, first out). Consider the situation where a thread performs a secure gateway to another process, and then yields and gets swapped out by the RTOS, saving its registers on the process stack. Next, a different thread performs a secure gateway to this same process, and it also yields and gets swapped out. Now, the RTOS wants to swap back in the original thread. The RTOS uses the stack pointer for that thread, and the registers are restored without any issue. The original thread then does a stack allocation that blaps over the saved state of the second thread, and the original thread then yields and is swapped out. Finally, the RTOS wants to swap the second thread back in. It uses the stack pointer for the second thread, but this pointer is now pointed to some "garbage" data leftover from the first thread's stack allocation.
```
Comment (AnotherButler) [inline docs/api/manual/UseCases.md]: What is a better word for "blaps"?

File: docs/api/manual/UseCases.md [hunk 4]
```diff
@@ -494,22 +475,22 @@ void unicorn_barf(unicorn_barfable_t thing);
 
 The function causes a unicorn to barf up some barfable thing (imaginarily, of course).
```
Comment (AnotherButler) [inline docs/api/manual/UseCases.md]: Please replace this example with a less graphic one.

File: docs/api/manual/UseCases.md [hunk 5]
```diff
@@ -308,97 +290,97 @@ After the receiver is done with the message, it must free resources backing the
...
+Secure gateway calls are synchronous, to avoid changing semantics. When the target process completes the function call on behalf of the caller, the source process receives the return code for the function call. Secure gateway calls cannot return complex types, such as structs.
 
 An asynchronous secure gateway is also introduced, to make asynchronous RPC easier than setting up a queue and using `uvisor_ipc_send`.
 -->
 
+## RTOS integration mechanics
 
-### RTOS Integration Mechanics
+As currently integrated with uVisor, the RTOS runs with uVisor privileges. This must be fixed. [This issue is tracked on GitHub](https://github.com/ARMmbed/uvisor/issues/235). For now, the RTOS is a privileged component of the overall system, sharing privileged residence with uVisor. uVisor and the RTOS must trust each other because of this. Generally, uVisor is the manager of interrupts and the MPU, and the RTOS is the manager of context switching.
```
Comment (AnotherButler) [inline docs/api/manual/UseCases.md]: This paragraph is confusing to me. Is it saying that RTOS has privileges but shouldn't? If so, why does the next paragraph list capabilities that support such integration? If not, what is it saying?

File: docs/api/manual/UseCases.md [hunk 6]
```diff
@@ -308,97 +290,97 @@ After the receiver is done with the message, it must free resources backing the
...
+  - IPC uVisor Secure Gateway Magic (to identify a potential IPC secure gateway).
+  - Source Box (Config Pointer).
+  - Target Box Queue Name.
+  - Target Box Namespace (optional).
 
-Secure gateway calls are synchronous, to avoid changing semantics. When the target process completes the function call on behalf of the caller, the source process receives the return code for the function call. Complex types (such as a struct) cannot be returned via a secure gateway call.
+Secure gateway calls are synchronous, to avoid changing semantics. When the target process completes the function call on behalf of the caller, the source process receives the return code for the function call. Secure gateway calls cannot return complex types, such as structs.
 
 An asynchronous secure gateway is also introduced, to make asynchronous RPC easier than setting up a queue and using `uvisor_ipc_send`.
 -->
```
Comment (AnotherButler) [inline docs/api/manual/UseCases.md]: Is this whole section supposed to be hidden? If so, should we find a way to hide it properly, so users can't see it?

File: docs/api/manual/UseCases.md [hunk 7]
```diff
@@ -54,46 +49,42 @@ typedef struct {
...
-page table memory must be owned by the calling security context.
+The allocator returns an error code if the page table is not formatted correctly. The requested `page_size` must be equal to `UVISOR_PAGE_SIZE`, and the calling security context must own all of the page table memory.
 
-Note that the returned pages are **not** guaranteed to be allocated consecutively.
-It is the responsibility of the tier-2 allocator to make sure that memory requests for continous memory, that exceed the requested page size, are blocked.
+Note that there is **no** guarantee of the consecutive allocation of returned pages. It is the responsibility of the tier-2 allocator to make sure that memory requests for continuous memory that exceed the requested page size are blocked.
 
 <!--
->>> Comment: This concept has been rejected for now, since this is prone to fragmentation.
+>>> Comment: This concept has been rejected for now because this is prone to fragmentation.
```
Comment (AnotherButler) [inline docs/api/manual/UseCases.md]: To what does this comment refer? Just the note above it? The whole section? If it doesn't work, should we take it out of the docs?

Comment (AlessandroA) [inline docs/api/manual/UseCases.md]: Yes it refers to the previous note. I'd keep it for now!

**Pull request merged**